src/DurableTask.Netherite/Connections/ConnectionInfoExtensions.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace DurableTask.Netherite
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using System.Net.Cache;
+    using System.Net.Http;
+    using System.Globalization;
+    using System.Security.Cryptography;
+    using System.Web;
+    using DurableTask.Netherite.EventHubsTransport;
+    using Microsoft.Azure.EventHubs;
+    using Azure.Core;
+    using System.Runtime.CompilerServices;
+    using Microsoft.Azure.EventHubs.Processor;
+    using Newtonsoft.Json.Serialization;
+    using DurableTask.Netherite.Faster;
+
+    /// <summary>
+    /// Utilities for constructing various SDK objects from a connection information.
+    /// </summary>
+    public static class ConnectionInfoExtensions
```

Comment:
Does this need to be `public`? With all the breaking change discussions in this PR, I recommend only making types `public` if they are expected to be consumed by end users.

If they need to be `public` for cross-assembly usage within this repo, we can use [internals-visible-to](https://github.com/microsoft/durabletask-dotnet/blob/main/eng/targets/InternalsVisibleTo.targets).

Reply:
These methods are useful if you want to slightly change something (e.g. the host URL) without having to rewrite all of it.

Reply:
I think they can be internal, yes.

Reply:
As helpful as providing extensions like this to customers is, I advise against it. Public API surface should be tightly controlled for maintainability's sake - each and every public API needs to have a good justification to be public. If customers think this would be helpful, they can copy the code into their own project.